be/src/olap/rowset/segment_v2/column_writer.cpp

```diff
@@ -142,8 +156,20 @@ Status ColumnWriter::_append_data(const uint8_t** ptr, size_t num_rows) {
             _null_bitmap_builder->add_run(false, num_written);
         }
 
-        // TODO(zc): update statistics for this page
-
+        if (_opts.need_zone_map) {
+            for (int i = 0; i < num_written; ++i) {
+                // TODO(hkp): optimize can be done here to reduce memcpy when the column is sorted
+                if (!_page_min_value->is_null()
+                        && _page_min_value->field()->compare(_page_min_value->cell_ptr(), start_ptr) > 0) {
```

Comment:
when input contains null, `!_page_min_value->is_null()` is false, which leads to the real not-null min value not being recorded

Reply:
ok